components/core/header/nav-bar/ScrollableNavBar.vue

```diff
@@ -0,0 +1,164 @@
+<template>
+    <div class="core-scrollableNavBar">
+        <button
+            class="core-scrollableNavBar__left-arrow"
+            @click="onLeftArrowClick"
+        >
+            <span
+                :class="{
+                    'core-scrollableNavBar__arrow-clickable': leftClickable,
+                }"
+            >
+                <fa :icon="['fa', 'chevron-left']" aria-hidden="true"></fa>
+            </span>
+        </button>
+        <button
+            class="core-scrollableNavBar__right-arrow"
+            @click="onRightArrowClick"
+        >
+            <span
+                :class="{
+                    'core-scrollableNavBar__arrow-clickable': rightClickable,
+                }"
+            >
+                <fa :icon="['fa', 'chevron-right']" aria-hidden="true"></fa>
+            </span>
+        </button>
+        <div
+            ref="scrollableNavBarScroll"
+            class="
+                core-scrollableNavBar__scroll core-scrollableNavBar__overflow
+            "
+        >
+            <div class="core-scrollableNavBar__slot">
+                <slot></slot>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+export default {
+    name: 'CoreScrollableNavBar',
+    props: {
+        scrollDelta: {
+            default: 1000,
+            type: Number,
+        },
+    },
+    data() {
+        return {
+            leftClickable: false,
+            rightClickable: true,
+        }
+    },
+    mounted() {
+        const el = this.$refs.scrollableNavBarScroll
+        this.leftClickable = !this.checkHorizontalScrollStart(el)
+        this.rightClickable = !this.checkHorizontalScrollEnd(el)
+    },
+    methods: {
+        checkHorizontalScrollStart(el, scrollLeft) {
+            if (scrollLeft === undefined) {
+                scrollLeft = el.scrollLeft
+            }
+            return scrollLeft <= 0
+        },
+        checkHorizontalScrollEnd(el, scrollLeft) {
+            if (scrollLeft === undefined) {
+                scrollLeft = el.scrollLeft
+            }
+            return el.scrollWidth - el.clientWidth - scrollLeft < 1
+        },
```

Comment:
Because we don't have strict ESLint settings, this is just a comment to point out a better practice.

Normally, in JavaScript, we don't reassign the parameter to avoid unexpected side effects. Instead, we make a new variable, e.g.,

```js
const predicate = scrollLeft || el.scrollLeft
return predicate <= 0
```

> Just a quick example, not a suggestion to change

After all, the methods are just doing a read-only kind of operation. In an edge case, if there's a global variable passed in, it'll become a [side effect](https://en.wikipedia.org/wiki/Side_effect_%28computer_science%29) which is unnecessary here.

Reply:
Because I try avoid the case like 0 || el.scrollLeft. But thanks for the good suggestion. Helped me remind the expression like 0 ?? el.scrollLeft. It's much cleaner, haha.